### PR TITLE
improvement(artifact): add scylla-doctor

### PR DESCRIPTION
**Pull request by juliayakovlev**
Add scylla-doctor to artifact test, run it and validate json output file.

Task:
https://github.com/scylladb/qa-tasks/issues/1617

### Testing

- [x] [artifacts-debian10-arm-test](https://argus.scylladb.com/test/55102838-2293-47d6-8bde-2ea1b27af077/runs?additionalRuns[]=6729e700-fc68-48a8-95a5-4f5cb8649084)
- [x] [artifacts-centos8-test](https://argus.scylladb.com/test/e38d1306-708f-4d21-b0ef-d22c492d697d/runs?additionalRuns[]=5b823a42-e6d7-4301-9356-2a0741842dee)
- [x] [Azure](https://argus.scylladb.com/test/302cc48f-8148-485c-b327-1469dea7f8cf/runs?additionalRuns[]=e41912ed-10b2-417a-a3ed-e18a826fb5f3) - ignore `FirewallRulesCollector` faliure - issue https://github.com/scylladb/field-engineering/issues/2248
- [x] [nonroot](https://argus.scylladb.com/test/a3b30dbd-667f-4d7e-bbdf-b5334eb0b600/runs?additionalRuns[]=e4c17b99-c00b-4312-935b-2c9c1222e577) - skip non-root test due to issue https://github.com/scylladb/field-engineering/issues/2254
- [x] [GCE image](https://argus.scylladb.com/test/2abd4416-f583-41e4-81cc-5d1ac98d7793/runs?additionalRuns[]=50b635c4-a5df-4976-aeb2-0850e6c0ec4c) - ignore `FirewallRulesCollector` faliure - issue https://github.com/scylladb/field-engineering/issues/2248
- [x] [flips](https://argus.scylladb.com/test/d024c8dd-5fe6-4c50-978e-eb27f13db930/runs?additionalRuns[]=340959d3-bb41-4c4b-8784-98e071ac8711) - skip test with client-encrypt due to [issue](https://github.com/scylladb/field-engineering/issues/2280)
- [x] [ami test](https://argus.scylladb.com/test/c29b1204-20a2-4c4c-9864-b0c47e80efc6/runs?additionalRuns[]=764e1a59-f7cc-4899-bab2-4617d6b46e5a) - ignore `FirewallRulesCollector` faliure - issue https://github.com/scylladb/field-engineering/issues/2248

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: sdcm/logcollector.py [hunk 1]
```diff
@@ -823,7 +823,9 @@ class ScyllaLogCollector(LogCollector):
                     CommandLog(name='scylla-manager-agent.yaml',
                                command=f'cat {SCYLLA_MANAGER_AGENT_YAML_PATH}'),
                     CommandLog(name='setup_scripts_errors.log',
-                               command='for i in /var/tmp/scylla/*.log;do echo [$i]; cat $i;done')
+                               command='for i in /var/tmp/scylla/*.log;do echo [$i]; cat $i;done'),
+                    CommandLog(name='scylla_doctor.vitals.json',
+                               command='cat *.vitals.json')
```
Comment (fruch) [inline sdcm/logcollector.py:828]: please add comma at the end:
```suggestion
                               command='cat *.vitals.json'),
```

Comment (juliayakovlev) [reply sdcm/logcollector.py]: done

**Comment by juliayakovlev** [comment]
@fruch which more tests worth to test?

File: utils/scylla_doctor.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+from argus.client.sct.types import Package
+
+from sdcm.cluster import BaseNode
+from sdcm.test_config import TestConfig
+
+LOGGING = logging.getLogger(__name__)
+
+
+class ScyllaDoctor:
+    SCYLLA_DOCTOR_LATEST_ON_AWS = "https://downloads.scylladb.com/downloads/scylla-doctor/latest"
```
[the remark this answers is not in the document]
Comment (juliayakovlev) [reply utils/scylla_doctor.py:16]: The task exists: https://github.com/scylladb/qa-tasks/issues/1682

**Comment by fruch** [comment]
Ok so we have follow up about nonroot and about client encryption...

But besides that seems like we are good to go (last few questions, and clarification above)

File: utils/scylla_doctor.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        self.node.install_package('wget')
+        self.node.remoter.run(f"wget {self.SCYLLA_DOCTOR_LATEST_ON_AWS}/{self.SCYLLA_DOCTOR_EXECUTOR}")
+        self.node.remoter.run(f"wget {self.SCYLLA_DOCTOR_LATEST_ON_AWS}/{self.SCYLLA_DOCTOR_CONF}")
+        self.scylla_doctor_exec = f"{self.current_dir}/{self.SCYLLA_DOCTOR_EXECUTOR}"
+        self.node.remoter.run(f"chmod 774 {self.scylla_doctor_exec}")
+        self.scylla_doctor_config = f"{self.current_dir}/{self.SCYLLA_DOCTOR_CONF}"
+
+        if self.node.is_nonroot_install:
+            self.python3_path = self.find_local_python3_binary(self.current_dir)
+            # TODO: update default paths
```
[the remark this answers is not in the document]
Comment (juliayakovlev) [reply utils/scylla_doctor.py:57]: I opened another task for nonroot test: https://github.com/scylladb/qa-tasks/issues/1683
Default paths in `scylla_doctor.conf` should be changed

File: sdcm/provision/helpers/certificate.py [hunk 4]
```diff
@@ -28,6 +28,8 @@ def install_client_certificate(remoter):
     setup_script = dedent("""
         mkdir -p ~/.cassandra/
         cp /tmp/ssl_conf/client/cqlshrc ~/.cassandra/
+        sudo mkdir /root/.cassandra
+        sudo cp /tmp/ssl_conf/client/cqlshrc /root/.cassandra/
```
[the remark this answers is not in the document]
Comment (juliayakovlev) [reply sdcm/provision/helpers/certificate.py]: removed

File: utils/scylla_doctor.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (fruch) [reply utils/scylla_doctor.py:57]: O.k.

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@juliayakovlev 

On Sunday take a look at the artifact tests in master and enterprise, if we have any fallbacks from this one.

**Pull request merged**